src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -31,7 +45,14 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
     });
   };
 
+  /**
+   * The options for the component
+   * @componentOptions
+   */
   static options: IQuerySuggestPreview = {
+    /**
+     * The result template to render each previews with.
```

Comment:
In PR https://github.com/coveo/search-ui/pull/1271, I added comments regarding this.

We need something more complete than this for our clients to be able to do something meaningful with this.

Reply:
I agree. Should I write it now or wait for @fbeaudoincoveo?

Reply:
Suggestion:

The HTML `id` attribute value, or CSS selector of the previously registered [result template](https://docs.coveo.com/413/) to apply when rendering the query suggestion result previews.

**Examples**

- Specifying the `id` attribute of the target result template:
   ```html
   <div class="CoveoQuerySuggestPreview" data-template-id="myTemplateId"></div>
   ```

- Specifying an equivalent CSS selector:
   ```html
   <div class="CoveoQuerySuggestPreview" data-template-selector="#myTemplateId"></div>
   ```

Reply:
Is there a default template?

Reply:
Just saw https://github.com/coveo/search-ui/pull/1271 hehe...

Let's add at the end of the description:

> If you specify no previously registered template through this option, the component uses a default template.

Reply:
I'll add the comment mentioning the default template to `JSUI-2695` given that it doesn't exist in `JSUI-2716` yet.

Reply:
Do we also document the fact that a template can be added as a child?
Also, do we give an example template? E.g.:
```html
<script class="result-template" type="text/html">
  <div class="coveo-result-frame">
    <div class="coveo-result-row">
      <div class="coveo-result-cell">
        <div class="CoveoImageFieldValue" data-field="@image" data-height="56"></div>
      </div>
    </div>
    <div class="coveo-result-row">
      <div class="coveo-result-cell"><a href="" class="CoveoResultLink"></a></div>
    </div>
  </div>
</script>
```